app/javascript/src/intro.js

```diff
@@ -4,66 +4,89 @@ import I18n from "./translations.js";
 
-  $("#menu-close").on("click", function (e) {
-    e.preventDefault();
-    $("#sidebar-wrapper").toggleClass("active");
-  });
+  setup_intl_tel_input();
+});
 
-  $(".alert .close").on("click", function () {
-    $(this).closest(".alert").remove();
+function setup_popup_close() {
+  $("#notice > .popup").on("click", function () {
+    $("#notice").fadeOut();
   });
+}
+
+function setup_auto_scroll_to_signup_error() {
+  const signupErrors = $("#signup-errors");
+
+  if (signupErrors.length > 0) {
+    const navbarOffset = $(".navbar").outerHeight();
+    const target = $("#enroll");
 
+    $("html,body").animate(
+      {
+        scrollTop: Math.ceil(Math.max(target.offset().top - navbarOffset, 0)),
+      },
+      1000,
+    );
+  }
+}
+
+function setup_smooth_scroll() {
   $("a[href*='#']").on("click", function () {
     if (
-      location.pathname.replace(/^\//, "") ==
+      location.pathname.replace(/^\//, "") ===
         this.pathname.replace(/^\//, "") ||
-      location.hostname == this.hostname
+      location.hostname === this.hostname
     ) {
-      var target = $(this.hash);
-      target = target.length ? target : $("[name=" + this.hash.slice(1) + "]");
-      if (target.length) {
+      const target = $(`${this.hash}, [name=${this.hash.slice(1)}]`);
+
+      if (target.length > 0) {
+        const navbarOffset = $(".navbar").outerHeight();
+
         $("html,body").animate(
           {
-            scrollTop: target.offset().top,
+            scrollTop: Math.ceil(
+              Math.max(target.offset().top - navbarOffset, 0),
+            ),
           },
           1000,
         );
         return false;
       }
     }
   });
+}
 
-  jQuery.validator.addMethod(
-    "valid_student_id",
-    function (value, element) {
-      if (/\F\d{6}/.test(value)) {
-        return true;
-      }
-      return true;
-    },
-    I18n.t("form.invalid_student_id"),
-  );
+function setup_background_parallax() {
+  let jumboHeight = $(window).height();
+  const headerBackground = $(".header-bg");
 
-  jQuery.validator.addMethod(
-    "required_if_minor",
-    function (value, element) {
-      // get max birth date to be 18 years old, ignoring time
-      var maxDate = new Date();
-      maxDate.setUTCHours(0, 0, 0, 0);
-      maxDate.setFullYear(maxDate.getFullYear() - 18);
+  $(window).on("resize", function () {
+    jumboHeight = $(window).height();
+  });
 
-      var birthDate = new Date($("#member_birth_date").val());
+  $(window).on("scroll resize", function () {
+    const scrolled = $(window).scrollTop();
+    headerBackground.css("height", jumboHeight - scrolled + "px");
+  });
+}
 
-      // field is valid if member is 18 years or if it's not empty
-      return birthDate < maxDate || value;
+function setup_form_validation() {
+  jQuery.validator.addMethod(
+    "phone",
+    function (value) {
+      return (
+        value.trim().length === 0 ||
+        /^\+?(?:[0-9] ?){6,14}[0-9]$/.test(value.trim())
```

Comment:
This regex matches a + sign. Is `value` here the full +31 6 69420 phone number, or just the text that was given by the user?

Reply:
During testing I could enter `+31 6 123456`. Will this be send to the server as `+31 +31 6 123456`? I could not find this that quickly (not a front-ender)

Reply:
The + sign is optional due to bootstrap; it will automatically add +31 (or whatever is selected) if it is not in the entered by the user. The original regex made it mandatory which messed with the system of the form. In my testing it works as expected. That being said, the server also checks the phone number and that is leading, so it is better if the server is more strict that the front-end.